core/store/models/eth.go

```diff
@@ -76,10 +86,32 @@ type Tx struct {
 	Hash      common.Hash
 	GasPrice  *Big `gorm:"type:varchar(255)"`
 	Confirmed bool
-	Hex       string `gorm:"type:text"`
 	SentAt    uint64
 }
 
+// String implements Stringer for Tx
+func (tx *Tx) String() string {
+	return fmt.Sprintf("Tx{ID: %d, From: %s, To: %s, Hash: %s, SentAt: %d}",
+		tx.ID,
+		tx.From.String(),
+		tx.To.String(),
+		tx.Hash.String(),
+		tx.SentAt)
+}
+
+// AfterCreate is used to add the default attempt to a TX after it is created
+// for the first time
+func (tx *Tx) AfterCreate(db *gorm.DB) (err error) {
```

Comment:
This gets automagically called by gorm?

Reply:
Yes it does, hence the removal of `CreateTxAndAttempt`, it also happens in the same transaction.